### DISCUSSION TITLE
Feature request: ability to specify and lock admin settings from command line switches or env

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#1476] [Q&A]
At PocketHost I’d like to lock SMTP settings via env vars (preferred) or CLI switches such that if it is specified that way, they cannot be changed. 

I’d prefer this over custom builds so I don’t need to maintain a build setup. 

Thoughts on how to do this? Or goja?

**Comment by ganigeorgiev** [comment]
At the moment I don't think this can be achieved without a custom build. The easiest way to lock settings changes would be to bind to the `OnSettingsBeforeUpdateRequest` hook:

```go
app.OnSettingsBeforeUpdateRequest().Add(func(e *core.SettingsUpdateEvent) error {
	log.Println(e.OldSettings.Smtp)
	log.Println(e.NewSettings.Smtp)

	// compare the settings fields you want to prevent changes and eventually return an error...
	if (...) {
		return apis.NewBadRequestError("Cannot change ... field.", nil)
	}

	return nil
})
```

Goja event hooks are not supported yet but I plan to implement them in v0.13/v0.14 (after the relation picker Admin UI improvements).
